src/util/piped_process.cpp

```diff
@@ -156,9 +156,9 @@ piped_processt::piped_processt(const std::vector<std::string> &commandvec)
     throw system_exceptiont("Input pipe creation failed for child_std_IN_Rd");
   }
   // Connect to the other side of the pipe
-  child_std_IN_Wr = CreateFile(
+  child_std_IN_Rd = CreateFile(
     in_name.c_str(),
-    GENERIC_WRITE,                                  // Write side
+    GENERIC_READ,                                   // Write side
```

Comment:
Does the comment here need updating as well?

Reply:
Does this even need a comment? I'd prefer to remove this comment so the we don't have to read and maintain the word write/read in two places.

Reply:
I have fixed this comment for the moment. I plan to remove the comment as part of a follow-up refactoring PR. I haven't included the removal in this PR in order to limit the scope of this PR to be _mostly_ bug fixing related.